### PR TITLE
[BYOC-DNNL] enable conv3d->bn folding

### PR DESCRIPTION
**Pull request by crazydemo**
This PR contains two modifications:

1. support conv3d-bn folding.
2. add some layout in dnnl_json_runtime, and fix some typos.

### CODE REVIEW

**Comment by masahi** [comment]
Is it possible to unify 2D and 3D impls?

Also please add test cases.

**Comment by crazydemo** [comment]
Is it possible to identify `conv2d` or `conv3d` from the given `callnode`?

**Comment by masahi** [comment]
yeah, for example 
https://github.com/apache/tvm/blob/4322ebc38797d8d76838db8be5918459ac3d0958/src/relay/backend/utils.h#L427-L432

**Comment by crazydemo** [comment]
> yeah, for example
> 
> https://github.com/apache/tvm/blob/4322ebc38797d8d76838db8be5918459ac3d0958/src/relay/backend/utils.h#L427-L432

Thank you so much! I will try to unify 2D and 3D impls.

**Comment by crazydemo** [comment]
Hi @masahi, would you please review this PR and give me some suggestions?

File: src/relay/transforms/fold_scale_axis.cc [hunk 1]
```diff
@@ -580,14 +581,38 @@ Expr Conv2DForwardRewrite(const Call& ref_call, const Array<Expr>& new_args,
...
+  return ConvForwardPrep(call, param, out_message);
+}
+
+Expr PreConvForwardRewrite(const Call& ref_call, const Array<Expr>& new_args,
+                           const Message& message) {
+  if (backend::IsOp(ref_call.as<CallNode>(), "nn.conv2d")) {
+    const auto* param = ref_call->attrs.as<Conv2DAttrs>();
+    return ConvForwardRewrite(ref_call, param, new_args, message);
+  }
+  const auto* param = ref_call->attrs.as<Conv3DAttrs>();
```
Comment (masahi) [inline src/relay/transforms/fold_scale_axis.cc:603]: `ICHECK(param)`

Add that to other places as well.

**Review by masahi — APPROVED** [review]
Sorry forgot.

I have one minor comment, please address it in the next PR.

File: src/relay/transforms/fold_scale_axis.cc (hunk 1 above)
Comment (crazydemo) [reply src/relay/transforms/fold_scale_axis.cc:603]: Got it.

**Pull request merged**